xbmc/filesystem/AddonsDirectory.cpp

```diff
@@ -807,9 +807,10 @@ void CAddonsDirectory::GenerateAddonListing(const CURL& path,
                                             const std::string label,
                                             bool alwaysShowUpdateIcon)
 {
-  std::set<std::shared_ptr<IAddon>> outdated;
-  for (const auto& addon : CServiceBroker::GetAddonMgr().GetOutdatedAddons())
-    outdated.insert(addon);
+  std::vector<std::shared_ptr<IAddon>> outdated;
+  std::vector<std::shared_ptr<IAddon>> updates;
+
+  CServiceBroker::GetAddonMgr().GetAvailableUpdatesAndOutdatedAddons(outdated, updates);
```

Comment:
Why not return a map of outdated addons -> updates. It should work ok with std::_any_of and is a more useful utility function than just combining two other functions.

Reply:
Actually, I think what you have right now is clean. If we need to return the relationship at a later point we can refactor this.